### PR TITLE
Add upload notification while there is an file upload going on

### PR DESCRIPTION
**Pull request by AlejandroRigau**
*Issue #, if available:* AIML-ADC-8952

*Description of changes:*
- Add upload notification while there is an file upload going on. Allows the user to navigate away while the upload is happening. 
- Add border radius to the drop-box to match radius of the containers. 

Demo:
Recording 2024-05-17 at 15 38 50

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by douglas1850 — DISMISSED** [review]

**Comment by AlejandroRigau** [comment]
it seems that I have a bug that makes the app crash if I refresh while file is uploading. Need to fix this first before getting approvals.

File: frontend/src/entities/dataset/dataset.css
Comment (dustins) [inline frontend/src/entities/dataset/dataset.css:33]: I don't know much about styling cloudscape styling, how sure are we that this variable is stable?

Comment (AlejandroRigau) [reply frontend/src/entities/dataset/dataset.css:33]: I spoke with the cloudscape team and it seems I should be using design tokens instead: https://cloudscape.design/foundation/visual-foundation/design-tokens/

**Comment by LightSeekerSC** [comment]
> it seems that I have a bug that makes the app crash if I refresh while file is uploading. Need to fix this first before getting approvals.

Keep in mind that this may just be a hard limitation. If you can't find a way to mitigate this, it is probably fine to put "please keep this page open and do not refresh as your files are uploading"

File: frontend/src/entities/dataset/dataset.service.tsx
Comment (LightSeekerSC) [inline frontend/src/entities/dataset/dataset.service.tsx:153]: Very cool! Love this! We may be able to use this in more places around the application.

**Comment by AlejandroRigau** [comment]
This is now ready for review

File: frontend/src/shared/layout/notification/notification.tsx
Comment (dustins) [inline frontend/src/shared/layout/notification/notification.tsx:34]: I actually think `useEffect` is more appropriate since you're not using the output. But I don't think it makes a difference.

**Review by dustins — APPROVED** [review]

File: frontend/src/shared/layout/notification/notification.tsx
Comment (AlejandroRigau) [reply frontend/src/shared/layout/notification/notification.tsx:34]: I'm using useMemo since it runs on first page load rather than running as a side effect of the page loading. It helps me fix some null errors when passing in components to notifications props.content

**Review by douglas1850 — APPROVED** [review]

**Pull request merged**
